src/wifi.rs

```diff
@@ -18,6 +18,11 @@ use esp_idf_hal::peripheral::Peripheral;
 
 use esp_idf_sys::*;
 
+#[cfg(feature = "async")]
```

Comment:
* The feature that you should be using is `nightly`, not `async`. Also see my other comments below, you only need to protect with `nightly` where you use a nightly feature. As in TAIT. Not here.